diff --git a/source/extensions/filters/http/ext_proc/ext_proc.cc b/source/extensions/filters/http/ext_proc/ext_proc.cc
index 8d4a62f0f8..08122227e7 100644
--- a/source/extensions/filters/http/ext_proc/ext_proc.cc
+++ b/source/extensions/filters/http/ext_proc/ext_proc.cc
@@ -342,7 +342,8 @@ Filter::StreamOpenState Filter::openStream() {
     auto options = Http::AsyncClient::StreamOptions()
                        .setParentSpan(decoder_callbacks_->activeSpan())
                        .setParentContext(grpc_context)
-                       .setBufferBodyForRetry(true);
+                       .setBufferBodyForRetry(true)
+                       .setSampled(absl::nullopt);
 
     ExternalProcessorStreamPtr stream_object =
         client_->start(*this, config_with_hash_key_, options, decoder_callbacks_);
