diff --git a/test/tools/router_check/router.cc b/test/tools/router_check/router.cc
index 410caf6cc0..84c9670d77 100644
--- a/test/tools/router_check/router.cc
+++ b/test/tools/router_check/router.cc
@@ -137,6 +137,9 @@ RouterCheckTool RouterCheckTool::create(const std::string& router_config_file,
   assignRuntimeFraction(route_config);
   auto factory_context =
       std::make_unique<NiceMock<Server::Configuration::MockServerFactoryContext>>();
+
+  Runtime::maybeSetDeprecatedInts("re2.max_program_size.error_level", 32768); // istio-proxy default
+
   auto config = *Router::ConfigImpl::create(route_config, *factory_context,
                                             ProtobufMessage::getNullValidationVisitor(), false);
   if (!disable_deprecation_check) {
